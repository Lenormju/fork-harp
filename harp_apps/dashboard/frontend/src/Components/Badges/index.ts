export { PerformanceRatingBadge } from "./PerformanceRatingBadge.tsx"
