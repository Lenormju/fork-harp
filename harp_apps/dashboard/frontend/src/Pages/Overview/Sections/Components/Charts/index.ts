export { TransactionsChart } from "./TransactionsChart.tsx"
