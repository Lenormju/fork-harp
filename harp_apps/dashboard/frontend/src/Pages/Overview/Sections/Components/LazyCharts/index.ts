import { lazy } from "react"

export const TransactionsChart = lazy(() => import("./TransactionsChart.tsx"))
export const SummaryPerformancesChart = lazy(() => import("./SummaryPerformancesChart.tsx"))
export const SummaryRateChart = lazy(() => import("./SummaryRateChart.tsx"))
