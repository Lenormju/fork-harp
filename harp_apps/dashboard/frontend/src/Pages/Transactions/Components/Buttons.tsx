import {
  ArrowPathIcon,
  ChevronDoubleLeftIcon,
  ChevronDoubleRightIcon,
  LinkIcon,
  XMarkIcon,
} from "@heroicons/react/16/solid"
import { ArrowDownIcon, ArrowUpIcon } from "@heroicons/react/24/outline"
import { HTMLAttributes } from "react"
import { Link } from "react-router-dom"
import tw, { styled } from "twin.macro"

interface ButtonProps {
  onClick?: () => unknown
}

const StyledButton = styled.button`
  ${tw`text-gray-400 mx-1 font-medium text-xs`}

  > svg {
    ${tw`h-3 w-3 inline-block`}
  }
`

export function VerticalFiltersShowButton({ onClick }: ButtonProps) {
  return (
    <StyledButton onClick={onClick}>
      <ChevronDoubleRightIcon />
      <div className="w-4">
        <div className="rotate-90">filters</div>
      </div>
    </StyledButton>
  )
}

export function FiltersHideButton({ onClick }: ButtonProps) {
  return (
    <StyledButton onClick={onClick}>
      <ChevronDoubleLeftIcon /> hide
    </StyledButton>
  )
}

export function FiltersResetButton({ onClick }: ButtonProps) {
  return (
    <StyledButton onClick={onClick}>
      <ArrowPathIcon /> reset
    </StyledButton>
  )
}

export function DetailsCloseButton({ onClick, ...moreProps }: ButtonProps & HTMLAttributes<HTMLButtonElement>) {
  return (
    <StyledButton onClick={onClick} {...moreProps}>
      <XMarkIcon /> close
    </StyledButton>
  )
}

export function PreviousButton({ onClick }: ButtonProps) {
  return (
    <StyledButton onClick={onClick}>
      <ArrowUpIcon /> previous
    </StyledButton>
  )
}

export function NextButton({ onClick }: ButtonProps) {
  return (
    <StyledButton onClick={onClick}>
      <ArrowDownIcon /> next
    </StyledButton>
  )
}

export function RefreshButton({ onClick, ...moreProps }: ButtonProps & HTMLAttributes<HTMLButtonElement>) {
  return (
    <StyledButton onClick={onClick} {...moreProps}>
      <ArrowPathIcon />
    </StyledButton>
  )
}

export function OpenInNewWindowLink({ id }: { id: string }) {
  return (
    <StyledButton as={(props) => <Link {...props} target="_blank" to={`/transactions/${id}`} />}>
      <LinkIcon /> open in new window
    </StyledButton>
  )
}
