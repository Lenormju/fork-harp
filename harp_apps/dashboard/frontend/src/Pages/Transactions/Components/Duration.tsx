import { XMarkIcon } from "@heroicons/react/16/solid"
import { CircleStackIcon } from "@heroicons/react/24/outline"
import { formatDuration } from "date-fns"

import ApdexBadge from "Components/Badges/ApdexBadge"
import { classNames } from "ui/Utilities"

export function NoCacheIcon() {
  return (
    <span className="h-4 w-4 relative">
      <CircleStackIcon className="absolute" />
      <XMarkIcon className="absolute" />
    </span>
  )
}

export const Duration = ({
  duration,
  apdex,
  cached = false,
  noCache = false,
  verbose = false,
  className = undefined,
}: {
  duration: number | null
  apdex: number | null
  cached?: boolean
  noCache?: boolean
  verbose?: boolean
  className?: string
}) => {
  if (duration !== null) {
    return (
      <div className={classNames("flex gap-x-0.5 items-center font-normal", className)}>
        {apdex !== null ? <ApdexBadge score={apdex} /> : null}
        <span>{formatDuration({ seconds: duration })}</span>
        {cached ? (
          <span className="h-4 flex text-xs text-slate-800" title="Response was cached.">
            <CircleStackIcon />
            {verbose ? "Cached" : null}
          </span>
        ) : null}
        {noCache ? (
          <span className="h-4 flex text-xs text-red-800" title="Cache was bypassed.">
            <NoCacheIcon />
            {verbose ? "Bypassed" : null}
          </span>
        ) : null}
      </div>
    )
  }
  return null
}
