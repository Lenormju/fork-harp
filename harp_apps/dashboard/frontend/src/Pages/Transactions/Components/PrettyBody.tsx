import SyntaxHighlighter from "react-syntax-highlighter/dist/esm/prism-light"
import BaseStyle from "react-syntax-highlighter/dist/esm/styles/prism/vs"

const Style = { ...BaseStyle }

export function PrettyBody({
  content = null,
  contentType = null,
}: {
  content: string | null
  contentType?: string | null
}) {
  switch (contentType) {
    case "application/json":
      return (
        <SyntaxHighlighter
          language="javascript"
          className="w-fit overflow-x-auto text-black language-javascript max-w-full text-xs"
          children={content || ""}
          style={Style}
          customStyle={{ padding: 0, border: 0 }}
        />
      )
  }

  return content ? (
    <pre className="max-w-full overflow-x-auto p-4 text-xs text-black" style={{ padding: 0, border: 0 }}>
      {content}
    </pre>
  ) : null
}
