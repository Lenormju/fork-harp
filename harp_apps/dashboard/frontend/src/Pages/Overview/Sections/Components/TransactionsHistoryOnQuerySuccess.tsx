import TpdexBadge from "Components/Badges/TpdexBadge.tsx"
import { OverviewData } from "Models/Overview"
import { H3 } from "ui/Components/Typography"

import { TransactionsChart } from "./LazyCharts"

export interface TransactionsHistoryOnQuerySuccessProps {
  data: OverviewData
  title?: string
  className?: string
}

export const TransactionsHistoryOnQuerySuccess = ({
  data,
  title,
  className,
}: TransactionsHistoryOnQuerySuccessProps) => {
  const meanDurationSeconds = Math.trunc(data.meanDuration) / 1000
  const meanTpdex = Math.trunc(data.meanTpdex)
  const errorsRate = data.errors.rate.toLocaleString(undefined, { style: "percent", minimumFractionDigits: 1 })
  return (
    <div className={className}>
      <div className="flex">
        <div className="flex flex-col max-w-60">
          <H3>{title}</H3>
          <div className="flex self-center">
            <TpdexBadge score={meanTpdex} size="xl">
              <div className="text-xs font-normal text-right">{data.timeRange} average</div>
            </TpdexBadge>
          </div>
          <div className="grid grid-cols-2 text-xs text-left align-text-bottom items-center self-center ml-10 mt-10">
            <span className="font-bold">Mean duration:</span>
            <span>{meanDurationSeconds} s</span>
            <span className="font-bold">Errors:</span>
            <span>{errorsRate}</span>
          </div>
        </div>
        <div className="grow h-full">
          <TransactionsChart data={data.transactions} timeRange={data.timeRange} />
        </div>
      </div>
    </div>
  )
}
