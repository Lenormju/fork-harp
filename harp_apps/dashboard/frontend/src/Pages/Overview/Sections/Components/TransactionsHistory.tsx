import { Loader } from "Components/Layout/Layout.tsx"
import { OnQuerySuccess } from "Components/Utilities/OnQuerySuccess.tsx"
import { useOverviewDataQuery } from "Domain/Overview"

import { TransactionsHistoryOnQuerySuccess } from "./TransactionsHistoryOnQuerySuccess.tsx"

interface TransactionsHistoryProps {
  endpoint?: string
  title?: string
  className?: string
  timeRange?: string
}

export const TransactionsHistory = ({ endpoint, title, className, timeRange }: TransactionsHistoryProps) => {
  const query = useOverviewDataQuery(endpoint, timeRange)

  return (
    <OnQuerySuccess query={query} queries={[query, query]} fallback={<Loader style={{ height: "300px" }} />}>
      {(query) => {
        return <TransactionsHistoryOnQuerySuccess data={query.data} title={title} className={className} />
      }}
    </OnQuerySuccess>
  )
}
