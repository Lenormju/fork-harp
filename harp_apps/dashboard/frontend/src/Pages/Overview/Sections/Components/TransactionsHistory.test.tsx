import { ErrorBoundary } from "react-error-boundary"
import { it } from "vitest"

import { Error } from "Components/Page"
import { renderWithClient } from "tests/utils.tsx"

import { TransactionsHistory } from "./TransactionsHistory.tsx"

it("renders the title and data when the query is successful", async () => {
  const result = renderWithClient(
    <ErrorBoundary FallbackComponent={Error}>
      <TransactionsHistory endpoint="test" title="Test Title" className="test-class" timeRange="month" />
    </ErrorBoundary>,
  )

  await result.findByText("Test Title")
  // todo see OverViewPage test
  //expect(result.container).toMatchSnapshot()
})
