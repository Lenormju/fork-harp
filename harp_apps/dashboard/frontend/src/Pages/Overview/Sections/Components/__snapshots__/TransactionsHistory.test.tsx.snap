// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the title and data when the query is successful 1`] = `
<div>
  <div
    class="test-class"
  >
    <h3
      class="css-1hpno4e"
    >
      Test Title
    </h3>
    <div
      style="display: flex; align-items: center;"
    >
      <div
        class="flex flex-col items-center"
      >
        <div
          class="flex self-center"
        >
          <span
            class="bg-green-500 css-1ubanwl"
            title="score: 95"
          >
            A
            <div
              class="text-xs font-normal text-right"
            >
              1h
               average
            </div>
          </span>
        </div>
        <div
          class="grid grid-cols-2 text-xs text-left align-text-bottom items-center self-center ml-10 mt-10"
        >
          <span
            class="font-bold"
          >
            Mean duration:
          </span>
          <span>
            0.03
             s
          </span>
          <span
            class="font-bold"
          >
            Errors:
          </span>
          <span>
            50,0Â %
          </span>
        </div>
      </div>
      <div
        class="recharts-responsive-container"
        style="width: 90%; height: 300px; min-width: 0;"
      />
    </div>
  </div>
</div>
`;
